File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 7/16/2026, 4:39:47 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        map <int ,int> res;

        for (int i = 0; i < n; i++){
            int rem = target - nums[i];
            if (res.find(rem) != res.end()){
                return {res[rem],i};
            };
            res[nums[i]] = i;
        }
        return {};
    }

};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 8/6/2026, 8:01:38 PM
1class Solution:
2    def twoSum(self, nums: List[int], target: int) -> List[int]:
3        L = {}
4        for i in range(len(nums)):
5            rem = target - nums[i]
6            if rem in L:
7                return [L[rem], i]
8            L[nums[i]] = i 
9